### ISSUE TITLE
[proposal] merge DOCs in one place

### ISSUE THREAD

**Issue by danikp** [ChicagoBoss/ChicagoBoss#618]
Currently we have 3 (three!!!!) places for documentation:
1) wiki
2) READMEs
3) API docs (produced from doc-src)
my proposal is to move everything to wiki. It will be pretty easy for READMEs as they written with markdown too. moving API docs will require much more work, but CB API is not so complicated and everything generated from code can be written once in markdown and changed on demand.

cos and pros:
* we will have one solid place for all CB documentation
* we will abandon some "offline" documentation, but, hey, does it's really important for web framework supposed to run online ?
As solution, with 'make doc' we can clone wiki repository and, optionally, convert markdown to html. So user will be required to be online only at that moment.

any other ideas welcome

**Comment by avellable** [comment]
+1

**Comment by JosKar** [comment]
+1

**Comment by choptastic** [comment]
Let me bounce in here for a moment with a thought:

TL;DR Version: It's my contention that the wiki should be left for long-form guides/walkthroughs/etc and the like, while the official documentation remains within the repo.

I really do like the idea of all documentation being in one place. But I'm not a fan of the github wiki.  It's quite nice that an update to the wiki can basically happen inline, but the wiki not being tied or synchronized with the codebase leads to what we already have -- a disconnect between the code and the wiki.

Say someone is making a PR with a new feature.  They would then have to update the wiki with the updated docs.  But what if the feature requires a bunch of changes, or whatever, then sits unused for a while and is finally merged a month or a year later (not unheard of).

When the PR finally happens, now it's time to update the documentation.  Normally, the best convention is to have the original contributor update the docs accordingly, but maybe said contributor doesn't have time right now, and waits a month (or never) to update the docs.  No one remembers to update the wiki, and now there's an undocumented feature.

This is the big problem with pulling the docs out of the repo itself.  Instead, the expectation should be that a new feature come packaged up with the documentation such that when the feature is finally merged, the docs update along with it.

I'd much rather see all documentation brought under the repo itself, including copying the (not-outdated) wiki contents into a main docs directory.

The side-bonus of that would be anyone who clones the repo will have access to the docs in their local repo without requiring connectivity.

**Comment by choptastic** [comment]
Ultimately, my biggest beef with the wiki is that it leads to a disconnect between the code and the documentation. With the code in the repo, updates to the docs happen within the same merge commit, and simplifies everyone's lives.  Then it's simple to say to a contributor: "Hey, this feature is great, after updating the documentation of this commit, I'll merge it in."

**Comment by danikp** [comment]
my current opinion is like that:  instead of extracting edoc definitions while processing doc-src we need to move all relevant information to files themselves. In such case API documentation could be build with './rebar doc' and all boss_doc machinery could be removed at all. So, developers can have API at their hand at any moment both inside a file they a looking at and as more fancy Edoc generated html files.

Everything else, like roadmap, READMEs and so on should be in wiki. Most of those things are not version specific and, in most cases, there is no need to update them if something changes in the code.
